File: app/src/main/java/com/bq/zowi/controllers/ProjectController.java
package com.bq.zowi.controllers;

import com.bq.zowi.models.Project;
import rx.Single;

/* JADX INFO: loaded from: classes.dex */
public interface ProjectController {
    public static final long PROJECT_QUIZ_BLOCKADE_MILLIS_ON_FAILURE = 600000;

    Single<Void> blockProjectQuiz(String str);

    Single<Project> getProject(String str);

    Single<Boolean> getProjectIsCompleted(String str);

    Single<Long> isProjectQuizBlocked(String str);

    Single<Void> resetAllProjects();

    Single<Void> setProjectCompleted(String str);
}

File: zowi-core/src/main/kotlin/com/bq/zowi/api/ProjectController.kt
package com.bq.zowi.api

import com.bq.zowi.models.Project
import io.reactivex.Single

interface ProjectController {
    companion object {
        const val PROJECT_QUIZ_BLOCKADE_MILLIS_ON_FAILURE = 600000L
    }

    fun blockProjectQuiz(projectId: String): Single<Void>
    fun getProject(projectId: String): Single<Project>
    fun getProjectIsCompleted(projectId: String): Single<Boolean>
    fun isProjectQuizBlocked(projectId: String): Single<Long>
    fun resetAllProjects(): Single<Void>
    fun setProjectCompleted(projectId: String): Single<Void>
}
